piet-direct2d/src/text.rs

```diff
@@ -127,18 +128,27 @@ impl TextLayoutBuilder for D2DTextLayoutBuilder {
         let layout = self.builder.build()?;
 
         let line_metrics = fetch_line_metrics(&layout);
+        let text_metrics = layout.get_metrics();
+        //FIXME: why is this including trailing whitespace?
```

Comment:
It probably should not. Basically the distinction is that for drawing and layout purposes, don't include trailing whitespace. The one case where including trailing whitespace is likely valid is cursor positioning, but conventions vary because positioning the cursor outside the layout rect can yield poor user experience.

Because both use cases are valid, a case can be made that we should have a metrics structure rather than a simple size. This metrics structure can contain other information such as the baseline of the first line. But I'm happy to consider that future work.